index.bs

```diff
@@ -2430,45 +2466,90 @@ To <dfn>check if an [=attribution source=] exceeds the time-based destination li
+1. [=set/iterate|For each=] [=event-level report=] |report| of the [=event-level report cache=]:
+    1. If |report|'s [=event-level report/trigger time=] is greater than or equal to |now|,
+        [=set/remove=] |report| from the [=event-level report cache=].
+
+        Note: This would only delete [=obtain a fake report|fake reports=] that would never be
+        stored if they were true [=event-level reports=].
+1. Let |deletedAggregatableReports| be a new [=set=].
+1. [=set/iterate|For each=] [=aggregatable report=] |report| of the [=aggregatable report cache=]:
+    1. If |sourcesToDelete| [=set/contains=] |report|'s [=aggregatable report/source identifier=]:
+        1. [=set/Append=] |report|'s [=attribution report/report ID=] to |deletedAggregatableReports|.
+        1. [=set/Remove=] |report| from the [=aggregatable report cache=].
+1. [=set/iterate|For each=] [=attribution rate-limit record=] |record| of the [=attribution rate-limit cache=]:
+    1. If |record|'s [=attribution rate-limit record/scope=] is:
+        <dl class="switch">
+        : "<code>[=rate-limit scope/source=]</code>"
+        :: Set |record|'s [=attribution rate-limit record/deactivated for unexpired destination limit=] to
+            true if |sourcesToDelete| [=set/contains=] |record|'s [=attribution rate-limit record/entity ID=].
+        : "<code>[=rate-limit scope/aggregatable-attribution=]</code>"
+        :: [=set/Remove=] |record| from the [=attribution rate-limit cache=] if
+            |deletedAggregatableReports| [=set/contains=] |record|'s [=attribution rate-limit record/entity ID=].
+
+        </dl>
+
+To <dfn>get sources to delete for the unexpired destination limit</dfn> given an
+[=attribution source=] |source|, run the following steps:
+1. Let |records| be a new [=list=].
+1. [=set/iterate|For each=] [=attribution rate-limit record=] |record| of the [=attribution rate-limit cache=]:
+    1. If |record|'s [=attribution rate-limit record/scope=] is not "<code>[=rate-limit scope/source=]</code>", [=iteration/continue=].
+    1. If |record|'s [=attribution rate-limit record/deactivated for unexpired destination limit=] is true, [=iteration/continue=].
+    1. If |record|'s [=attribution rate-limit record/source site=] and |source|'s [=attribution source/source site=] are not equal, [=iteration/continue=].
+    1. If |record|'s [=attribution rate-limit record/reporting origin=] and |source|'s [=attribution source/reporting origin=] are not [=same site=], [=iteration/continue=].
+    1. If |record|'s [=attribution rate-limit record/expiry time=] is less than or equal to |source|'s [=attribution source/source time=], [=iteration/continue=].
+    1. [=Assert=]: |record|'s [=attribution rate-limit record/destination limit priority=] is not null.
+    1. [=Assert=]: |record|'s [=attribution rate-limit record/entity ID=] is not null.
+    1. [=list/Append=] the [=tuple=] (|record|'s [=attribution rate-limit record/destination limit priority=],
```

Comment:
IMO this tuple has enough fields that it would be better to use a struct with named fields.

Reply:
Updated with a struct, thanks.